File: 1-two-sum/1-two-sum.cpp

class Solution {
public:
    
        int linearsearch(vector<int> nums, int start, int x)
        {
            for(int i = start; i < nums.size(); i++)
                if(nums[i] == x)
                    return i;
            return -1;
        }
        vector<int> twoSum(vector<int>& nums, int target) 
        {
            vector<int> x = nums;        
            sort(x.begin(), x.end());
            int a = 0, b = x.size() - 1;

            while(a < b)
            {
                if(x[a] + x[b] == target)
                    break;
                if(x[a] + x[b] > target)
                    b--;
                else
                    a++;
            }
            a = x[a];
            b = x[b];
            int a_i = linearsearch(nums, 0, a);
            int b_i = 0;
            if(a == b)
                b_i = linearsearch(nums, a_i + 1, b);
            else
                b_i = linearsearch(nums, 0, b);
            vector <int> ans = {a_i, b_i};
            return ans;
        }
};

File: 1-TwoSum/1-TwoSum.py
# Last updated: 8/30/2025, 8:21:51 PM
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash_map = {}
        for i in range(len(nums)):
            hash_map[nums[i]] = i
        for i in range(len(nums)):
            complement = target - nums[i]
            if complement in hash_map and hash_map[complement] != i:
                return [i, hash_map[complement]]
        # If no valid pair is found, return an empty list
        return []